src/app/services/abuse-ip.service.ts

```diff
@@ -0,0 +1,21 @@
+import { Injectable } from '@angular/core';
+import { HttpClient } from '@angular/common/http';
+import { environment } from '../../environments/environment';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AbuseIpService {
+  private abuseIpUrl = environment.abuseIpUrl;
+  private abuseIpApiKey = environment.abuseIpApiKey;
+
+  constructor(private httpClient: HttpClient) {}
+
+  public check(ipAddress: string) {
+    return this.httpClient.get(`${this.abuseIpUrl}/check/${ipAddress}?maxAgeInDays=90`, {
```

Comment:
Does Angular's HTTP client support a base URL? Just out of curiosity.

Reply:
There is a way by creating an interceptor: https://stackoverflow.com/questions/45735655/how-do-i-set-the-baseurl-for-angular-httpclient